### DISCUSSION TITLE
Feature request: support perf or other sampling profilers

### DISCUSSION THREAD

**Discussion by FedericoCeratto** [nextest-rs/nextest#1795] [Feature requests]
### Motivation

Sometimes tests are slow and running a sampling profiler can be useful.

### Proposal

Support running a sampling profile around a test and let it generate an output file.

**Comment by sunshowers** [comment]
Thanks for the request! This is currently possible with [target runners](https://nexte.st/docs/features/target-runners/), but I agree more in-box functionality would be nice. In general, I think it makes sense to ship reasonably common target runners:

* sampling profilers like perf
* debuggers like gdb
* tracers like strace and (d)truss
* and in general, an easy way to say "insert some kind of binary or script between nextest and the actual test itself", with various options for things like capturing vs passing through stdout/stderr, writing output to files that can be discovered later, etc.

It's a bit of daunting design work to be honest though, and I'm interested in hearing your general thoughts on what might be useful for you.
